### PR TITLE
[Docs] Japanese translation of internals_*.md

### PR DESCRIPTION
**Pull request by umi-umi**
## Description

Add Japanese translation of
- docs/internals_defines.md
- docs/internals_midi_device.md
- docs/internals_midi_device_setup_process.md
- docs/internals_midi_util.md
- docs/internals_send_functions.md
- docs/internals_sysex_tools.md

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by mtei** [comment]
@shelaf @s-show @hsgw @satt99 @takashiski @yoshihara よかったらレビューお願いします。

File: docs/ja/internals_send_functions.md [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+#### パラメータ
+* `device` 送信に使うデバイス
+
+* `b` 送信するバイト
+
+#### `public void `[`midi_send_data`](#group__send__functions_1ga36e2f2e45369d911b76969361679054b)`(`[`MidiDevice`](#struct__midi__device)` * device,uint16_t count,uint8_t byte0,uint8_t byte1,uint8_t byte2)` {#group__send__functions_1ga36e2f2e45369d911b76969361679054b}
+
+最大3バイトのデータを送信する。
+
+数える時に4の剰余が適用されるため、これを使って sysex を渡すことができます。
```
Comment (shelaf) [inline docs/ja/internals_send_functions.md:221]: countはパラメータ名を指しているように思います
```suggestion
これを使って sysex を通過できるように count に4の剰余が適用されます。
```

**Review by shelaf — COMMENTED** [review]
全体的にですが、メンバーの方で概要にない説明がですます調で追加されているので、概要もですます調にすると全体が揃うように思います。

File: docs/ja/internals_midi_device.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+デバイスのプロセス関数の最初に呼び出される関数が必要な場合、midi_device_set_pre_input_process_func を使います。通常、入力をポーリングして midi_device_input に渡します。
+
+## 概要
+
+メンバー                        | 説明                                        
+--------------------------------|---------------------------------------------
+`define `[`MIDI_INPUT_QUEUE_LENGTH`](#group__midi__device_1ga4aaa419caebdca2bbdfc1331e79781a8)            | 
+`enum `[`input_state_t`](#group__midi__device_1gac203e877d3df4275ceb8e7180a61f621) | 
+`public void `[`midi_device_input`](#group__midi__device_1gad8d3db8eb35d9cfa51ef036a0a9d70db)`(`[`MidiDevice`](#struct__midi__device)` * device,uint8_t cnt,uint8_t * input)` | 入力バイトを処理する。この関数はバイトを解析し、指定されたデバイスに関連する適切なコールバックを呼びます。この関数は、カスタムデバイスを作成していて、MIDI 入力を取得したい場合に使います。
+`public void `[`midi_device_set_send_func`](#group__midi__device_1ga59f5a46bdd4452f186cc73d9e96d4673)`(`[`MidiDevice`](#struct__midi__device)` * device,midi_var_byte_func_t send_func)` | 出力データバイトの送信に使われるコールバック関数を設定します。これは、カスタムデバイスを作成している場合のみ使われます。多くの場合、コールバック関数で割り込みを無効にして、ロックを気にすることなく様々な MIDI 送信関数を呼び出すことができます。
```
[the remark this answers is not in the document]
Comment (umi-umi) [reply docs/ja/internals_midi_device.md]: ご指摘ありがとうございます。

File: docs/ja/internals_sysex_tools.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+#### パラメータ
+* `encoded` 出力データバッファは、少なくとも sysex_encoded_length(length) バイトの長さが必要です。
+
+* `source` エンコードされるデータの入力バッファ。
+
+* `length` 入力バッファからエンコードするバイト数。
+
+#### 返り値
+number of bytes encoded.
```
[the remark this answers is not in the document]
Comment (umi-umi) [reply docs/ja/internals_sysex_tools.md]: ご指摘ありがとうございます。

**Review by shelaf — APPROVED** [review]

**Review by mtei — APPROVED** [review]

**Comment by mtei** [comment]
Thank you!

**Comment by fauxpark** [comment]
IMO this should be reverted as these files are technically generated automatically.

**Comment by mtei** [comment]
What you mean is that we shouldn't translate these documents?

**Comment by fauxpark** [comment]
Yes. The original docs are generated with doxygen, however it needs a little work (and actual doc comments in the code) to generate more than just the MIDI stuff here.

**Comment by mtei** [comment]
First, those documents don't seem to have been updated for nearly two years.

```
commit ae7284edb8bd8ed7fcab29415049bf03baf57f8d
Author: Jack Humbert <jack.humb@gmail.com>
Date:   Tue Mar 27 03:24:30 2018

    rename api docs to internals

M       docs/_summary.md
R100    docs/api_defines.md     docs/internals_defines.md
R100    docs/api_input_callback_reg.md  docs/internals_input_callback_reg.md
R100    docs/api_midi_device.md docs/internals_midi_device.md
R100    docs/api_midi_device_setup_process.md   docs/internals_midi_device_setup_process.md
R100    docs/api_midi_util.md   docs/internals_midi_util.md
R100    docs/api_send_functions.md      docs/internals_send_functions.md
R100    docs/api_sysex_tools.md docs/internals_sysex_tools.md
```

Second, what do I need to do to automatically generate a Japanese version of the document?

Third, how long should a Japanese-speaking reader wait to get a Japanese document? 

I think it's okay to translate the current version as it is now. Even if it's an unfinished document. In the future, if someone prepares to automatically generate a multilingual document, I'll be happy to do that and redo the translation.

**Comment by fauxpark** [comment]
> First, those documents don't seem to have been updated for nearly two years.

As I said, the doxygen stuff needs work.

> Second, what do I need to do to automatically generate a Japanese version of the document?

It should not be translated at all, because it's generated based on the doc comments. It's a waste of time.

https://www.doxygen.nl/index.html

**Comment by fauxpark** [comment]
Looks like there is some i18n support, but I'm not seeing a whole lot of green: https://www.doxygen.nl/manual/langhowto.html

**Comment by mtei** [comment]
Do you want to say is that we should make the source files in the following manner?

```c
/**
 * English comments
 * 日本語のコメント
 * 中文评论
 * 中文評論
 * 한국어 코멘트 
 */
int foo_bar(void);
```

This is difficult to maintain.

Comments should be written in one language. It is practical to translate the generated resulting document into each language.

**Comment by mtei** [comment]
@qmk/collaborators 
Does anyone have an opinion?

**Comment by fauxpark** [comment]
> Do you want to say is that we should make the source files in the following manner?

No, what I'm saying is that it should not be manually translated. We can generate these docs in other languages, but it will only be the doxygen template strings.

**Comment by zvecr** [comment]
I vote revert. Right now the doxygen comments provide little value, and I think we should also keep the comments to English only.

**Comment by hsgw** [comment]
I think I'd better translate it.
There are many people in the Japanese community who are not that familiar with either English or programming.
They will not be able to tell if this is an automatically generated document or not.

**Comment by s-show** [comment]
For people in the Japanese community who are not familiar with English or programming, the official reference of QMK translated into Japanese is a great help.
For such Japanese, it is a trivial matter whether these documents are automatically generated or written by humans.
I am translating official references for those people, as are translators other than me.
Therefore, the results of this translation should remain until the original document is modified.

**Pull request merged**